modules/library/main/src/main/java/org/geotools/filter/function/EqualIntervalFunction.java

```diff
@@ -220,4 +242,66 @@ public RangedClassifier evaluate(Object object) {
         }
         return calculate((SimpleFeatureCollection) object);
     }
+
+    private double[] getNotNumericalPercentages(int classNum, int totalSize) {
+        int lastBigBin = totalSize % classNum;
+        if (lastBigBin != 0) lastBigBin--;
+        double classMembers = (double) totalSize / classNum;
+        double[] percentages = new double[classNum];
+        for (int i = 0; i < classNum; i++) {
+            percentages[i] = (classMembers / totalSize) * 100;
+            if (lastBigBin != 0 && lastBigBin == i) {
+                classMembers--;
+            }
+        }
+        return percentages;
+    }
+
+    private double[] getNumericalPercentages(
+            int classNum, RangedClassifier classifier, FeatureCollection collection)
+            throws IOException {
+        double max = ((Number) classifier.getMax(classifier.getSize() - 1)).doubleValue();
+        double min = ((Number) classifier.getMin(0)).doubleValue();
+        double classWidth = (max - min) / classNum;
+        Subtract subtract = FF.subtract(getParameters().get(0), FF.literal(min));
+        Divide divide = FF.divide(subtract, FF.literal(classWidth));
+
+        GroupByVisitor groupBy =
+                new GroupByVisitor(
+                        Aggregate.COUNT, getParameters().get(0), Arrays.asList(divide), null);
+        collection.accepts(groupBy, null);
+        Object[] result = groupBy.getResult().toArray();
```

Comment:
Normally I would get the result as a map, why turn it into an array? The way the result is given also seems bizarre, is it an array of arrays?

Reply:
fixed